src/smtpclient.js

```diff
@@ -548,11 +558,33 @@
 
 
     SmtpClient.prototype._send = function(buffer) {
-        var timeout = this.TIMEOUT_SOCKET_LOWER_BOUND + Math.floor(buffer.byteLength * this.TIMEOUT_SOCKET_MULTIPLIER);
+        this._setTimeout(buffer.byteLength);
+        return this.socket.send(buffer);
+    };
+
+    SmtpClient.prototype._setTimeout = function(byteLength) {
+        var prolongPeriod = Math.floor(byteLength * this.TIMEOUT_SOCKET_MULTIPLIER);
+        var timeout;
+
+        if (this._dataMode) {
+            // we're in data mode, so we count only one timeout that get extended for every send().
+            var now = Date.now();
+
+            // the old timeout start time
+            this._socketTimeoutStart = this._socketTimeoutStart || now;
+
+            // the old timeout period, normalized to a minimum of TIMEOUT_SOCKET_LOWER_BOUND
+            this._socketTimeoutPeriod = (this._socketTimeoutPeriod || this.TIMEOUT_SOCKET_LOWER_BOUND) + prolongPeriod;
```

Comment:
I think that `_socketTimeoutPeriod` and `_socketTimeoutStart ` should be cleared somewhere as well, otherwise you might run into issues if you want to send several messages during one session (for newer messages old timeout values are used)